### DISCUSSION TITLE
Is it possible to share a custom next.js server between two next.js apps?

### DISCUSSION THREAD

**Discussion by tytremblay** [vercel/next.js#35483] [Help]
I'd like to have the following directory structure:
```
client1
   - pages 
client2
  - pages
server
  - server.ts
```

`server.ts` is an express server where I've configured the next server according to [this documentation](https://nextjs.org/docs/advanced-features/custom-server).  I've defined two next apps and set the `dir` option accordingly:

```
const app1 = next({dir: '../client1'})
const app2 = next({dir: '../client2'})
```

Routing seems to work, but I receive an error about an invalid hook call when the page tries to render.  Both clients are vanilla `create-next-app` projects at the moment.  

Is my configuration incorrect?  Or is this not supposed to be done in the first place and I should work on a refactor / different framework?

**Comment by GabenGar** [comment]
This is probably what you are looking for:
https://nextjs.org/docs/advanced-features/multi-zones

**Reply by tytremblay** [reply]
Thank you.  This reads like two zones is two independent Next.js that coexist.  In this instance, wouldn't there be two independent Next.js servers running as well?

**Reply by GabenGar** [reply]
Next.js app is a server by itself, so what did you mean by "two next.js apps" then? Also zones are not independent, one of the zones is designated as the main one and uses rewrites in the config.
